burn-derive/src/record/base.rs

```diff
@@ -22,12 +22,19 @@ struct RecordDeriveCodegen {
     name_item: Ident,
     gen: StructRecordItemCodegen,
     generics: Generics,
+    has_backend: bool,
```

Comment:
Is this done for legacy? I am not sure when Backend would not be present

Reply:
This is intended for record with and without the backend trait defined as generic. Since the `Record` trait didn't have any generic before, we didn't have to handle that case.

```rust
#[derive(Record)]
struct NoBackend;

// We need to generate the following

impl<B: Backend> Record<B> for NoBackend {
   ...
}

#[derive(Record)]
struct MyModule<B: Backend> {
  layer: nn::Linear<B>
}

// We need to generate the following, but we don't need to manually add the backend trait, since it's already defined.

impl<B: Backend> Record<B> for MyModule<B> {
   ...
}
```